lib/bootloader_setup.pm

```diff
@@ -318,7 +318,7 @@ sub get_bootmenu_console_params {
 sub uefi_bootmenu_params {
     # assume bios+grub+anim already waited in start.sh
     # in grub2 it's tricky to set the screen resolution
-    send_key "e";
+    send_key_until_needlematch('grub2-enter-edit-mode', 'e', 5, 0.5);
```

Comment:
I've merged the needles, but I guess it will be almost impossible to catch all the scenarios where it's used. So not to break others tests, I would propose either make this optional and just press `e` for all other cases as before, or announce it to all the groups, so that we can create all the needles in advance.
Another point, for the magic numbers you have derived empirically, does it worth it to write wrapper for it, similarly to `type_string_slow` we have to `type_string`?

Reply:
I am afraid that the o3 tests won't be even able to get to this point since there is a tiny chance to assert canceled GRUB2 timeout as it is for SLES. 
http://eris.suse.cz/tests/17418#step/bootloader_uefi/2

The default grub2 timeout on JeOS images is set to 10s, which is quite neck-breaking in openQA. But in general, I am counting that not so many UEFI compliant jobs, thus it should not be such an impossible task.
In the worst-case scenario, I can still create a dedicated module for JeOS uefi boot with respect to the JeOS specific GRUB2 setup.

Reply:
Or you could use variables to ignore grub and let system boot, then disable grub timeout and reboot.  @asmorodskyi as you worked with it, what was the variable name to ignore grub and wait till the system boots?

Reply:
KEEP_GRUB_TIMEOUT=1

Reply:
That sounds very nice! Thanks, guys! I will try it out since we are setting only the GRUB2 screen resolution. Other than that, we do not need to interfere with GRUB2

Reply:
ok, I cannot skip gfxpayload setup http://eris.suse.cz/tests/17438#step/firstrun/2

Reply:
> Another point, for the magic numbers you have derived empirically, does it worth it to write wrapper for it, similarly to `type_string_slow` we have to `type_string`?

Eventually, I created jeos specific grub2 handler module where *send_key_until_needlematch* part will be executed only on JeOS. As you stated before, it won't make sense to try to apply it globally.